13.go

```diff
@@ -1032,12 +1041,21 @@ func (hs *clientHandshakeState) doTLS13Handshake() error {
 		return unexpectedMessageError(certMsg, msg)
 	}
 	hs.keySchedule.write(certMsg.marshal())
+
 	// Validate certificates.
 	certs := getCertsFromEntries(certMsg.certificates)
 	if err := hs.processCertsFromServer(certs); err != nil {
 		return err
 	}
 
+	// Validate the delegated credential if present.
+	if len(certMsg.certificates) > 0 {
```

Comment:
The current is condition is necessary, since `len(certMsg.certificates)` could be `0` at this point. Verifying that the client indicated DCs is checked by `processDelegatedCredentialFromServer()`. I did it this way because this check is needed here and at `handshake_client.go:482`.

Added a comment clarifying this behavior.

Reply:
Right. I ment that this condition shouldn't be entered when AcceptDeletegatedCredential is set to false (so both len()>0 and AcceptDeletegatedCredential=true). But indeed, that's done.  All good here

Reply:
Ack.